src/services/pallets/PalletsErrorsService.ts

```diff
@@ -0,0 +1,124 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ApiDecoration } from '@polkadot/api/types';
+import { Text } from '@polkadot/types';
+import {
+	BlockHash,
+	ErrorMetadataLatest,
+	PalletErrorMetadataV14,
+} from '@polkadot/types/interfaces';
+import { IsError } from '@polkadot/types/metadata/decorate/types';
+import { stringCamelCase } from '@polkadot/util';
+import { IPalletErrors, IPalletErrorsItem } from 'src/types/responses';
+
+import { AbstractPalletsService } from '../AbstractPalletsService';
+
+interface IFetchPalletArgs {
+	hash: BlockHash;
+	palletId: string;
+}
+
+interface IFetchErrorItemArgs extends IFetchPalletArgs {
+	errorItemId: string;
+	metadata: boolean;
+}
+
+export class PalletsErrorsService extends AbstractPalletsService {
+	async fetchErrorItem(
+		historicApi: ApiDecoration<'promise'>,
+		{ hash, palletId, errorItemId, metadata }: IFetchErrorItemArgs
+	): Promise<IPalletErrorsItem> {
+		const metadataFieldType = 'errors';
+		const palletMetadata = historicApi.registry.metadata;
+
+		const [palletMeta, palletMetaIdx] = this.findPalletMeta(
+			palletMetadata,
+			historicApi,
+			palletId,
+			metadataFieldType
+		);
+
+		// Even if `errorItemMeta` is not used, we call this function to ensure it exists. The side effects
+		// of the error item not existing are that `findErrorItemMeta` will throw.
+		const errorItemMetadata = this.findPalletFieldItemMeta(
+			historicApi,
+			palletMeta,
+			errorItemId,
+			metadataFieldType
+		) as PalletErrorMetadataV14;
+
+		let palletErrorMetadata: ErrorMetadataLatest | undefined;
+		if (metadata) {
+			palletErrorMetadata = (errorItemMetadata[1] as IsError).meta;
+		}
+
+		const { number } = await this.api.rpc.chain.getHeader(hash);
+
+		return {
+			at: {
+				hash: hash,
+				height: number.unwrap().toString(10),
+			},
+			pallet: stringCamelCase(palletMeta.name),
+			palletIndex: palletMetaIdx,
+			errorItem: errorItemId,
+			metadata: palletErrorMetadata,
+		};
+	}
+
+	async fetchErrors(
+		historicApi: ApiDecoration<'promise'>,
+		{ hash, palletId, onlyIds }: IFetchPalletArgs & { onlyIds: boolean }
+	): Promise<IPalletErrors> {
+		const metadataFieldType = 'errors';
+		const metadata = historicApi.registry.metadata;
+		const [palletMeta, palletMetaIdx] = this.findPalletMeta(
+			metadata,
+			historicApi,
+			palletId,
+			metadataFieldType
+		);
+
+		const { number } = await this.api.rpc.chain.getHeader(hash);
+		const errors = historicApi.errors[palletMeta.name.toString().toLowerCase()];
```

Comment:
While connected in Kusama, I tried to query some endpoints (to get the errors for some pallets) like : 
- `http://127.0.0.1:8080/pallets/42/errors/` -> [FastUnstake Pallet](https://github.com/paritytech/polkadot/blob/master/runtime/kusama/src/lib.rs#L1428)
- `http://127.0.0.1:8080/pallets/15/errors/` -> [TechnicalCommittee](https://github.com/paritytech/polkadot/blob/master/runtime/kusama/src/lib.rs#L1358)

and it gives me the following error : 

```
{
  "code": 500,
  "message": "Cannot convert undefined or null to object",
  "stack": "TypeError: Cannot convert undefined or null to object\n    at Function.entries (<anonymous>)\n    at PalletsErrorsService.fetchErrors (/Users/tiny_imod/Documents/Parity/substrate-api-sidecar/build/src/services/pallets/PalletsErrorsService.js:60:44)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async PalletsErrorsController.getErrors (/Users/tiny_imod/Documents/Parity/substrate-api-sidecar/build/src/controllers/pallets/PalletsErrorsController.js:54:56)\n    at async /Users/tiny_imod/Documents/Parity/substrate-api-sidecar/build/src/controllers/AbstractController.js:226:9",
  "level": "error"
}

```
I noticed that the error appears when the const variable `errors` stays `undefined` after this statement (in line 97) is executed.

However, when I query other pallets like 
- `http://127.0.0.1:8080/pallets/5/errors/` -> [authorship](https://github.com/paritytech/polkadot/blob/master/runtime/kusama/src/lib.rs#L1346) 

I get the expected results (list of pallet errors) and I saw that for those endpoints the const variable `errors` has values (the expected pallet errors) after this statement is executed.

Reply:
great catch! Seems I had a bit of logic to lowercase the pallet name in the fetchErrors function and the pallets in tests were a single word such as `democracy` so cases for something like fastUnstake weren't covered. Ive removed the lower casing of palletName values and added  e2e test for polkadots `PhragmenElection` pallet in order to cover this case.

Reply:
That works now! Great! :)